src/Analyzers/VisualBasic/Tests/Iterator/IteratorTests.vb

```diff
@@ -73,90 +89,101 @@ End Module")
 
         <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToIterator)>
         Public Async Function TestConvertToIteratorSubLambda() As Task
-            Await TestMissingInRegularAndScriptAsync(
+            Await VerifyNoCodeFixAsync(
 "Imports System
 Imports System.Collections.Generic
 
 Module Module1
     Sub M()
         Dim a As Func(Of IEnumerable(Of Integer)) = Sub()
-                                                        [|Yield|] 0 
+                                                        {|BC30451:Yield|} {|BC30800:1 |}
  End Sub
     End Sub
 End Module")
         End Function
 
         <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToIterator)>
         Public Async Function TestConvertToIteratorSingleLineFunctionLambda() As Task
-            Await TestMissingInRegularAndScriptAsync(
+            Await VerifyNoCodeFixAsync(
 "Imports System
 Imports System.Collections.Generic
 
 Module Module1
     Sub M()
-        Dim a As Func(Of IEnumerable(Of Integer)) = Function() [|Yield|] 0 
+        Dim a As Func(Of IEnumerable(Of Integer)) = Function() {|BC30451:Yield|} {|BC30205:1|} 
  End Sub
 End Module")
         End Function
 
         <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToIterator)>
         Public Async Function TestConvertToIteratorSingleLineSubLambda() As Task
-            Await TestMissingInRegularAndScriptAsync(
+            Await VerifyNoCodeFixAsync(
 "Imports System
 Imports System.Collections.Generic
 
 Module Module1
     Sub M()
-        Dim a As Func(Of IEnumerable(Of Integer)) = Sub() [|Yield|] 0 
+        Dim a As Func(Of IEnumerable(Of Integer)) = Sub() {|BC30451:Yield|} {|BC30800:1 |}
  End Sub
 End Module")
         End Function
     End Class
 
     Public Class ChangeToYieldTests
-        Inherits AbstractVisualBasicDiagnosticProviderBasedUserDiagnosticTest
+        Private Shared Async Function VerifyCodeFixAsync(code As String, fixedCode As String) As Task
+            Await New VerifyConvertToYield.Test With
+            {
+                .TestCode = code,
+                .FixedCode = fixedCode
```

Comment:
You specified `.CodeActionValidationMode = CodeActionValidationMode.None` in the prior test class, but not here. Any specific reason? What does `CodeActionValidationMode.None` do?

Reply:
@mavasani The testing library, by default, verifies the syntax tree exactly matches the one produced by the compiler. I used `CodeActionValidationMode.None` to compare text only because the codefix doesn't produce the same tree.